### ISSUE TITLE
Latest builds don't seem to properly sync information

### ISSUE THREAD

**Issue by JarheadHME** [Tuesday1028/GTFO_AccuracyTracker#4]
It's possible you are already aware of this. I just want to check.

A friend and I both use this plugin, and are using the build I made with the changes discussed in #3 . Because these are very minor changes and they don't write any accuracy data, nor touch the events, I believe this likely exists in the current `main` branch.

# My in-game tracker
![4Upd4CT](https://github.com/Tuesday1028/GTFO_AccuracyTracker/assets/15879336/256001fc-08bb-4109-940e-40f163902167)
# My end screen
![NcVC4MU](https://github.com/Tuesday1028/GTFO_AccuracyTracker/assets/15879336/9fcf6ef6-ac37-477a-9bbe-d71533063a16)

# Friend's in-game tracker
![image](https://github.com/Tuesday1028/GTFO_AccuracyTracker/assets/15879336/b25ae912-20e9-48e9-9fcb-8bec3e1f8453)
# Friend's end screen
![image](https://github.com/Tuesday1028/GTFO_AccuracyTracker/assets/15879336/dc265b70-0d30-41da-b82c-eb9316e37a7d)

**Comment by JarheadHME** [comment]
It seems like, in the `pAccuracyData` constructor, `Owner.SetPlayer(player);` doesn't actually update the `pPlayer`. 
Removing the ` = new();` from the line declaring `Owner`  and instead setting `Owner` like so in the constructor seems to work:
`Owner = new() { lookup = player.Lookup, IsBot = player.IsBot, };`

In summary:
```diff
public struct pAccuracyData
    {
        internal pAccuracyData(SNet_Player player, Dictionary<InventorySlot, AccuracyData.AccuracySlotData> slotDatas)
        {
-            Owner.SetPlayer(player);
+            Owner = new()
+            {
+                lookup = player.Lookup,
+                IsBot = player.IsBot,
+            };
            if (slotDatas.TryGetValue(InventorySlot.GearStandard, out var standardSlotData))
            {
                StandardSlotData = new(standardSlotData);
            }
            if (slotDatas.TryGetValue(InventorySlot.GearSpecial, out var specialSlotData))
            {
                SpecialSlotData = new(specialSlotData);
            }
        }

-        public SNetStructs.pPlayer Owner { get; private set; } = new();
+        public SNetStructs.pPlayer Owner { get; private set; }
        public pAccuracySlotData StandardSlotData { get; private set; } = new();
        public pAccuracySlotData SpecialSlotData { get; private set; } = new();
    }
```

**Comment by JarheadHME** [comment]
Played with it tonight, and I'm not really sure how, but this happened. ![image](https://i.imgur.com/DQHUg8j.png) 
(I'm green. I'm also host. I had shot things by this point.)
After this level, I restarted my game. The next level we did, it only showed my own accuracy, and no one else's. It still kept my information at 0/0/0 on the display. 
![image](https://i.imgur.com/YJ5hb7D.png)
The following levels after that, the display didn't show at all. 
![image](https://i.imgur.com/gK5VffY.png)

Interestingly, for all of these levels, it still tracked our accuracy and displayed it on the end screen. Exactly how accurate it was, I don't know for sure. Strangely, For those first two rounds, on the end screen it displayed my accuracy information with the * icon, seemingly indicating that I didn't have the mod. 
![image](https://i.imgur.com/dfd3ad1.png)
